code/controllers/subsystem/daynightcycle.dm

```diff
@@ -41,27 +41,36 @@ var/datum/subsystem/daynightcycle/SSDayNight
 		flags |= SS_NO_FIRE
 		pause()
 	else
-		switch(timeOfDay)
+		
+		time2fire()
+		
+		switch(next_timeOfday)
 			if(TOD_MORNING)
-				timeOfDay = TOD_DAYTIME
+				next_light_power = 10
```

Comment:
shouldn't this go under TOD_NIGHTTIME? otherwise it won't turn back on for morning

Reply:
I thought you wanted 50% of a 24 hour day to have no light?